tests/Entity/BaseEntityManagerTest.php

```diff
@@ -16,74 +16,99 @@
 
-        $this->getManager()->exception;
+        $this->manager->exception;
     }
 
-    public function testExceptionOnNonMappedEntity()
+    /**
+     * Test throw exception on non mapped entity.
+     */
+    public function testExceptionOnNonMappedEntity(): void
     {
         $this->expectException(\RuntimeException::class);
         $this->expectExceptionMessage('Unable to find the mapping information for the class classname. Please check the `auto_mapping` option (http://symfony.com/doc/current/reference/configuration/doctrine.html#configuration-overview) or add the bundle to the `mappings` section in the doctrine configuration');
 
-        $registry = $this->createMock(ManagerRegistry::class);
-        $registry->expects($this->once())->method('getManagerForClass')->willReturn(null);
+        $this->registry->expects($this->once())->method('getManagerForClass')->willReturn(null);
 
-        $manager = new EntityManager('classname', $registry);
-        $manager->getObjectManager();
+        $this->manager->getObjectManager();
     }
 
-    public function testGetEntityManager()
+    /**
+     * Test get entity manager.
+     */
+    public function testGetEntityManager(): void
     {
-        $objectManager = $this->createMock(ObjectManager::class);
-
-        $registry = $this->createMock(ManagerRegistry::class);
-        $registry->expects($this->once())->method('getManagerForClass')->willReturn($objectManager);
-
-        $manager = new EntityManager('classname', $registry);
+        $this->registry->expects($this->once())->method('getManagerForClass')->willReturn($this->objectManager);
 
-        $manager->em;
+        $this->manager->em;
     }
 
+    /**
+     * Test get repository by class.
+     *
+     * @throws \ReflectionException
+     */
     public function testGetRepository(): void
     {
         $entityRepository = $this->createMock(EntityRepository::class);
-        $objectManager = $this->createMock(ObjectManager::class);
-        $objectManager->expects($this->once())->method('getRepository')->with('classname')->willReturn($entityRepository);
 
-        $registry = $this->createMock(ManagerRegistry::class);
-        $registry->expects($this->once())->method('getManagerForClass')->willReturn($objectManager);
+        $this->objectManager->expects($this->once())->method('getRepository')->with('classname')->willReturn($entityRepository);
+
+        $this->registry->expects($this->once())->method('getManagerForClass')->willReturn($this->objectManager);
+
+        $r = new \ReflectionObject($this->manager);
+        $m = $r->getMethod('getRepository');
+        $m->setAccessible(true);
 
-        $manager = new EntityManager('classname', $registry);
-        $repository = $manager->getRepositoryFromBaseClass();
-        $this->assertInstanceOf(EntityRepository::class, $repository);
+        $this->assertInstanceOf(EntityRepository::class, $m->invoke($this->manager));
```

Comment:
Why do we have to change all this test exactly? I don't see how this relates to your code change…

Reply:
Ok I think I get it now, you're using reflection because of `getMockForAbstracClass()`, allowing you to call the actual code. I guess that's fine.